paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/PaymentMethodCommonSpec.kt

```diff
@@ -1,38 +1,46 @@
 package com.stripe.android.paymentsheet.forms
 
+import androidx.annotation.StringRes
+import androidx.compose.ui.graphics.Color
+import com.stripe.android.paymentsheet.forms.FormElementSpec.SectionSpec
+
 
 /**
  * This class is used to define different forms full of fields.
  */
 data class FormSpec(
-    val layout: FieldLayoutSpec,
+    val layout: Layout,
     val paramKey: MutableMap<String, Any?>,
-) {
-    val allTypes get() = layout.sections.map { it.field }
-}
+)
 
 /**
  * This is a data representation of the layout of UI fields on the screen.
  */
-data class FieldLayoutSpec(val sections: List<SectionSpec>) {
-    val allFields get() = sections.map { it.field }
+data class Layout(val elements: List<FormElementSpec>) {
+    val allFields
+        get() = elements.filterIsInstance<SectionSpec>().map { it.field }
 }
 
 /**
  * This is used to define each section in the visual form layout
  */
-data class SectionSpec(
-    val field: SectionFieldSpec
-) {
-    sealed class SectionFieldSpec(
-        // This is the key used in the PaymentMethodCreateParams
-        val identifier: String,
-    ) {
-        object Name : SectionFieldSpec("name")
-
-        object Email : SectionFieldSpec("email")
-
-        object Country : SectionFieldSpec("country")
+
+sealed class FormElementSpec {
+    data class SectionSpec(val field: SectionFieldSpec) : FormElementSpec() {
+        sealed class SectionFieldSpec(val identifier: String) {
+            object Name : SectionFieldSpec("name")
+
+            object Email : SectionFieldSpec("email")
+
+            object Country : SectionFieldSpec("country")
+        }
+    }
+
+    sealed class StaticSpec : FormElementSpec() {
```

Comment:
This seems a data structure that adds a text field to `SectionFieldSpec`, why is it called Static?

Reply:
I can remove this additional layer of hierarchy, as I was having discussions it was helpful to distinguish between a text field and static text, but I think I can do this just by calling it StaticTextSpec.

Reply:
I will make changes